### PR TITLE
Bioconductor 3.7 iomkl 2018a R 3.5.0

### PR DESCRIPTION
**Pull request by edmondac**
This follows on from #6466 - which needs to be merged first. This will then just contain two easyconfigs.

Extensions updated with easy_update.

### CODE REVIEW

**Comment by edmondac** [comment]
I'm not sure what to do about "found 2 variants of 'R' dependency in easyconfigs using '2018a' toolchain generation"...

**Comment by edmondac** [comment]
@boegel - here's a PR for Bioconductor 3.7, but it's falling foul of the multivariant deps test. Any suggestions?

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 13 out of 19 (2 easyconfigs in this PR)
nic21 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/2ba39b1245fad6fae10e5ecf5f6af411 for a full test report.

**Comment by boegel** [comment]
@edmondac Sorry for not getting back to you yet on this...

We should enhance the test for it to allow easyconfigs that are specific to a particular version of R (like we already do with CUDA for example).

**Comment by boegel** [comment]
@edmondac Problem with test should be fixed with https://github.com/bear-software/easybuild-easyconfigs/pull/2

**Comment by edmondac** [comment]
Thanks @boegel :-)

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb [hunk 1]
```diff
@@ -0,0 +1,937 @@
...
+
+dependencies = [
+    ('R', '3.5.0', '-X11-20180131'),
+    ('HDF5', '1.10.1'),  # for rhdf5
+    ('MariaDB-connector-c', '2.3.5'),  # for RMySQL
+]
+
+exts_default_options = {
+    'source_urls': [
+        'https://bioconductor.org/packages/3.7/bioc/src/contrib/',
```
Comment (boegel) [inline easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb]: @edmondac download is failing for `Rsamtools`, probably because the version was updated

This can be mitigated by updating the `Rsamtools` version (current is `1.32.2`), but also by including the archive source URL, cfr. https://github.com/easybuilders/easybuild-easyconfigs/pull/6311/files

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb [hunk 2]
```diff
@@ -0,0 +1,937 @@
...
+    ('futile.logger', '1.4.3', {
+        'checksums': ['5e8b32d65f77a86d17d90fd8690fc085aa0612df8018e4d6d6c1a60fa65776e4'],
+    }),
+    ('snow', '0.4-2', {
+        'checksums': ['ee070187aea3607c9ca6235399b3db3e181348692405d038e962e06aefccabd7'],
+    }),
+    ('BiocParallel', '1.14.1', {
+        'checksums': ['d6c8c76de48fa0f3e2e81abac0a3e39b205719080917aea4bf18272ade0c0f03'],
+    }),
+    ('Rsamtools', '1.32.0', {
```
Comment (boegel) [inline easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb]: please update to `1.32.2`

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb [hunk 3]
```diff
@@ -0,0 +1,937 @@
...
+        'checksums': ['c4f80de2f2233c58fc658548293f8cdec3dfe76bda0e27457ce284df3dfff196'],
+    }),
+    ('RBGL', '1.56.0', {
+        'checksums': ['0e5a056a1a63e711bf73ec9978b525dfb7adb49f63e01cba2bd15f5cab384942'],
+    }),
+    ('org.Hs.eg.db', '3.6.0', {
+        'checksums': ['27c838a6d343e9cb869d20dacf489fc0e3eac7a881a8bc2840a3a0ad10235b49'],
+    }),
+    ('XML', '3.98-1.11', {
+        'checksums': ['505bd948152213c22e255b8d38b303c0fac5e113c0f9b1bcc9bfa62e1edd3fa2'],
```
Comment (boegel) [inline easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb]: checksum failed, I'm getting `90c83af4b1a304d04f97eaf0c5b651b02135dc15cce7e4e38d732d48ae9e30b8` on a fresh download...

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb (hunk 1 above)
Comment (edmondac) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb]: I've added the archive line

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb (hunk 2 above)
Comment (edmondac) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb]: Done

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb (hunk 3 above)
Comment (edmondac) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb]: Deleted my existing download and ran inject-checksums

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
node2001.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/60886937fb7b49cb8fe65c364a300cfd for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2446.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/daf7fdf3967f8718fe0c8187030b5fea for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2001.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/22cc8bf18ba9d11f878e25f680ff99a5 for a full test report.

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb [hunk 4]
```diff
@@ -0,0 +1,938 @@
...
+    ('promises', '1.0.1', {
+        'checksums': ['c2dbc7734adf009377a41e570dfe0d82afb91335c9d0ca1ef464b9bdcca65558'],
+    }),
+    ('httpuv', '1.4.4.1', {
+        'checksums': ['c6a44f262df83bae0fcf808a07c0b3da5f0bc9ea29aa7c6eccf421d541c07c8a'],
+    }),
+    ('sourcetools', '0.1.7', {
+        'checksums': ['47984406efb3b3face133979ccbae9fefb7360b9a6ca1a1c11473681418ed2ca'],
+    }),
+    ('shiny', '1.1.0', {
```
Comment (boegel) [inline easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb]: @edmondac There are a whole bunch of additional R libraries included in here... Why is that necessary, these are already provided as a part of the R 3.5.0 dependency?

Comment (edmondac) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb]: Hmm. I didn't add them deliberately. Presumably easy_update added them.

Comment (edmondac) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.7-iomkl-2018a-R-3.5.0.eb]: I'll remove them

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2596.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/3570350ac41b5de9f1ec872ff0f80340 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @edmondac!
